File: python/0701二叉搜索树的插入..py
from binarytree import TreeNode

def insertIntoBST(root: TreeNode, val: int) -> TreeNode:
    if(not root):
        return TreeNode(val)
    
    if(val>root.val):
        root.right=insertIntoBST(root.right,val)
    
    else:
        root.left=insertIntoBST(root.left,val)
    
    return root

File: cpp/0701二叉搜索树的插入.cpp
#include"mybitree.h"

TreeNode *insertIntoBST(TreeNode *root, int val)
{
    if(!root)
        return new TreeNode(val);

    TreeNode *temp = root;
    while (temp)
    {
        if (val > temp->val)
        {
            if (!temp->right)
            {
                temp->right = new TreeNode(val);
                break;
            }
            temp = temp->right;
        }

        // val不会与BST中节点相等
        else
        {
            if (!temp->left)
            {
                temp->left = new TreeNode(val);
                break;
            }
            temp = temp->left;
        }
    }
    return root;
}